journalpump/rsyslog.py

```diff
@@ -133,11 +149,30 @@ def close(self):
             self.socket = None
 
     def send(self, message):
-        if self.socket is None:
-            self._connect()
-        self.socket.sendall(message[:self.max_msg - 1])
-        if len(message) >= self.max_msg:
-            self.socket.sendall(b"\n")
+        attempts = 0
+        while True:
+            try:
+                if self.socket is None:
+                    self._connect()
+
+                self.socket.sendall(message[:self.max_msg - 1])
+                if len(message) >= self.max_msg:
+                    self.socket.sendall(b"\n")
+
+                break
+            except Exception as ex:  # pylint: disable=broad-except
+                self.close()
+                attempts += 1
+
+                if attempts > self.retry_attempts or not self._should_retry(ex=ex):
+                    raise
+
+                time.sleep(self.retry_backoff_seconds)
+
+    def _should_retry(self, *, ex):
+        if isinstance(ex, OSError):
+            return ex.errno is errno.EPIPE
```

Comment:
`is` => `==`, otherwise this emits SyntaxWarning

Reply:
Thanks!